### PR TITLE
Eos 22542 : Btree code changes for capturing transactions for KV Insert operation.

### PR DESCRIPTION
**Pull request by RadhaGulhane13**
- Added P_CAPTURE phase will capture nodes in transaction.

### CODE REVIEW

File: btree/btree.c [hunk 1]
```diff
@@ -3703,6 +3710,21 @@ static int64_t btree_put_kv_tick(struct m0_sm_op *smop)
...
+		int                  i;
+
+		for (i = 0; i < BTREE_CALLBACK_CREDIT; i++) {
+			if (arr[i].nc_node == NULL)
+				break;
+			node_slot.s_node = arr[i].nc_node;
+			node_slot.s_idx  = arr[i].nc_idx;
+			node_capture(&node_slot, bop->bo_tx);
+		}
+
```
Comment (NikhilBirgade) [inline btree/btree.c:3727]: It looks the change in `del_kv_tick` for P_CAPTURE are missing.

File: btree/btree.c [hunk 2]
```diff
@@ -1517,6 +1518,11 @@ static void node_move(struct nd *src, struct nd *tgt,
 	tgt->n_type->nt_move(src, tgt, dir, nr, tx);
 }
 
+static void node_capture(struct slot *slot, struct m0_be_tx *tx)
```
Comment (NikhilBirgade) [inline btree/btree.c:1521]: Structure name and function name are same (`node_capture`). Please change one of these.

File: btree/btree.c (hunk 1 above)
Comment (RadhaGulhane13) [reply btree/btree.c:3727]: It needs to get cover in different ticket EOS-22550, as some more changes are required for capturing transaction for Del operation.

File: btree/btree.c [hunk 3]
```diff
@@ -1212,7 +1213,7 @@ struct level {
  * This stucture will store the address of node descriptor and index of record
  * which will be used for capturing transaction.
  */
-struct node_capture {
+struct node_to_capture {
```
Comment (shashank-parulekar) [inline btree/btree.c:1216]: s/node_to_capture/node_capture_info/

File: btree/btree.c (hunk 2 above)
Comment (RadhaGulhane13) [reply btree/btree.c:1521]: Changed structure name to node_to_capture

File: btree/btree.c [hunk 4]
```diff
@@ -3044,11 +3050,11 @@ static void level_cleanup(struct m0_btree_oimpl *oi, struct m0_be_tx *tx)
 /**
  * Adds unique node descriptor address to m0_btree_oimpl::i_capture structure.
  */
-static void btree_callback_add(struct m0_btree_oimpl *oi, struct nd *addr,
+static void btree_node_capture(struct m0_btree_oimpl *oi, struct nd *addr,
```
Comment (shashank-parulekar) [inline btree/btree.c:3053]: s/btree_node_capture/btree_node_capture_enlist/

**Review by shashank-parulekar — CHANGES_REQUESTED** [review]

**Review by NikhilBirgade — APPROVED** [review]

**Review by shashank-parulekar — APPROVED** [review]

**Pull request merged**